src/Fantomas.Core/CodePrinter.fs

```diff
@@ -3412,7 +3412,9 @@ and genMultilineSimpleRecordTypeDefn astContext openingBrace withKeyword ms ao'
 
 and genMultilineSimpleRecordTypeDefnAlignBrackets astContext openingBrace withKeyword ms ao' fs closingBrace =
     // the typeName is already printed
-    opt (indent +> sepNln) ao' genAccess
+    ifStroustrupElse
+        (opt (sepNlnWhenWriteBeforeNewlineNotEmpty sepSpace) ao' genAccess)
```

Comment:
The `sepNlnWhenWriteBeforeNewlineNotEmpty` will not really play out when there is a comment after the access modifier.

Something like:
```fsharp
type Person = private // foo
    {
        FirstName: string
        LastName: string
    }
```

would be formatted to

```fsharp
type Person = private // foo
{
    FirstName: string
    LastName: string
}
```

It is better to use `sepSpace` in this case and let the comment flow after the `{` like

```fsharp
type Person = private { // foo
    FirstName: string
    LastName: string
}
```

This isn't super accurate but good enough for now.

Reply:
Should I add a test case that covers this scenario?

Reply:
I would like you to change `(sepNlnWhenWriteBeforeNewlineNotEmpty sepSpace)` with `sepSpace` here.

Reply:
> I would like you to change `(sepNlnWhenWriteBeforeNewlineNotEmpty sepSpace)` with `sepSpace` here.

Right, just asking if you'd like me to add an additional test case or not.

Reply:
No, it's fine for now. Adding the test case described above wouldn't strictly be correct.
If you have something like `private // foo` after formatting that part should in theory be preserved. This is now not the case but the output is acceptable, so I would leave it be.

Reply:
Alright, I've made the requested changes and removed another redundant case.